File: lab4 mpp/TeledonPersistence/DonatieDBRepository.cs
using teledon.domain;
using log4net;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace teledon.repository
{
    public class DonatieDBRepository : IDonatieRepo
    {
        private static readonly ILog log = LogManager.GetLogger("RepoDonatie");
        private DonatorDBRepository donatorRepo = new DonatorDBRepository();
        private CazDBRepository cazRepo = new CazDBRepository();


        public Donatie Delete(long id)
        {
            log.InfoFormat("entering delete cazuri with value {0}", id);
            IDbConnection con = DBUtils.getConnection();

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "delete from Cazuri where idCaz=@id";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@id";
                paramId.Value = id;
                comm.Parameters.Add(paramId);
                var dataR = comm.ExecuteNonQuery();

            }
            log.InfoFormat("exiting delete cazuri");
            return null;
        }

        public IEnumerable<Donatie> FindAll()
        {
            throw new NotImplementedException();
        }

        public Donatie FindOne(long id)
        {
            throw new NotImplementedException();
        }

        public Donatie Save(Donatie entity)
        {
            IDbConnection con = DBUtils.getConnection();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "Insert into Donatii(idDonator, idCaz, suma)  values (@idDonator, @idCaz, @suma)";

                IDbDataParameter paramIdDonator = comm.CreateParameter();
                paramIdDonator.ParameterName = "@idDonator";
                paramIdDonator.Value = entity.Don.ID;
                comm.Parameters.Add(paramIdDonator);

                IDbDataParameter paramIdCaz = comm.CreateParameter();
                paramIdCaz.ParameterName = "@idCaz";
                paramIdCaz.Value = entity.Caz.ID;
                comm.Parameters.Add(paramIdCaz);

                IDbDataParameter paramSuma = comm.CreateParameter();
                paramSuma.ParameterName = "@suma";
                paramSuma.Value = entity.Suma;
                comm.Parameters.Add(paramSuma);


                var result = comm.ExecuteNonQuery();

                if (result == 0)
                    log.InfoFormat("save exited with value {0}", result);
                else
                    log.InfoFormat("save exited with value {0}", result);

            }


            return null;
        }

        public Donatie Update(Donatie entity)
        {
            throw new NotImplementedException();
        }
    }
}

File: GUImpp/TeledonPersistence/src/main/java/teledon/persistence/repository/DonatieDBRepository.java
package teledon.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import teledon.model.CazCaritabil;
import teledon.model.Donatie;
import teledon.model.Donator;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class DonatieDBRepository implements IDonatieRepo {
    private JdbcUtils dbUtils;

    private static final Logger logger = LogManager.getLogger();
    private CazDBRepository cazRepo;
    private DonatorDBRepository donatorRepo;

    public DonatieDBRepository(Properties properties) {
        logger.info("Initializing DonatorDBRepository with properties: {} ", properties);
        dbUtils = new JdbcUtils(properties);
        donatorRepo = new DonatorDBRepository(properties);
        cazRepo = new CazDBRepository(properties);
    }

    @Override
    public Donatie findOne(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        Donatie donatie = new Donatie();
        try(PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Donatii WHERE idDonatie=?")) {
            preparedStatement.setLong(1, aLong);
            ResultSet result = preparedStatement.executeQuery();
            int id = result.getInt("idDonatie");
            Donator donator = donatorRepo.findOne(result.getLong("idDonator"));
            CazCaritabil caz = cazRepo.findOne(result.getLong("idCaz"));
            int suma = result.getInt("suma");
            donatie = new Donatie(donator, caz, suma);
            donatie.setId(aLong);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " +e);
        }
        logger.traceExit(donatie);
        return donatie;
    }

    @Override
    public Iterable<Donatie> findAll() {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Donatie> donatii = new ArrayList<>();

        try (PreparedStatement preparedStatement = con.prepareStatement("SELECT * FROM Donatii")) {
            try(ResultSet result = preparedStatement.executeQuery()) {
                while(result.next()) {
                    Long id = result.getLong("idDonatie");
                    Donator donator = donatorRepo.findOne(result.getLong("idDonator"));
                    CazCaritabil caz = cazRepo.findOne(result.getLong("idCaz"));
                    int suma = result.getInt("suma");
                    Donatie donatie = new Donatie(donator, caz, suma);
                    donatie.setId(id);
                    donatii.add(donatie);

                }
            }
        } catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(donatii);
        return donatii;
    }

    @Override
    public Donatie save(Donatie entity) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("insert into Donatii(idDonator, idCaz, suma) values (?,?,?)")){
            preparedStatement.setInt(1, entity.getDonator().getId().intValue());
            preparedStatement.setInt(2, entity.getCaz().getId().intValue());
            preparedStatement.setInt(3, entity.getSuma());
            int result = preparedStatement.executeUpdate();
            logger.trace("Saved {} instances", result);
        }
        catch(SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return entity;
    }

    @Override
    public Donatie delete(Long aLong) {
        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        try(PreparedStatement preparedStatement = con.prepareStatement("DELETE FROM Donatii WHERE idDonatie = ?")){
            preparedStatement.setInt(1, aLong.intValue());
            int result = preparedStatement.executeUpdate();
            logger.trace("Deleted {} instances", result);
        }
        catch(SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        return null;
    }

    @Override
    public Donatie update(Donatie entity) {
        return null;
    }
}
